webfrontend/src/components/setup/avatar.tsx

```diff
@@ -0,0 +1,191 @@
+import "bootstrap/dist/css/bootstrap.min.css";
+import AvatarEditor from "react-avatar-editor";
+import { Button, Form, Image } from "react-bootstrap";
+import { triggerErrorMessage } from "../../utils/locals";
+import { postRequestContent } from "../../utils/request";
+import Modal from "react-bootstrap/Modal";
+
+interface ImageProperties {
+  originalImage: string | File;
+  croppedImage: string | undefined;
+  position: { x: number; y: number };
+  scale: number;
+  rotate: number;
+}
+
+interface Props {
+  currentProfileImg: string | File;
+  setIsEditProfileImg: React.Dispatch<React.SetStateAction<boolean>>;
+  isEditProfileImg: boolean;
+  setProfileImgUrl: React.Dispatch<React.SetStateAction<File | undefined>>;
+}
+
+const Avatar: React.FC<Props> = ({
+  setIsEditProfileImg,
+  isEditProfileImg,
+  setProfileImgUrl,
+  currentProfileImg,
+}) => {
+  const editorRef: React.RefObject<AvatarEditor> = createRef();
+  const [imgToBeSubmitted, setImgToBeSubmitted] = useState<File>();
+  const [fileName, setFileName] = useState("");
+  const [imageProperties, setImageProperties] = useState<ImageProperties>({
+    originalImage: currentProfileImg,
+    croppedImage: undefined,
+    position: { x: 0.5, y: 0.5 },
+    scale: 1,
+    rotate: 0,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { originalImage, croppedImage, position, scale, rotate } =
+    imageProperties;
+
+  const handleAdd = useCallback((event: React.ChangeEvent<any>) => {
+    setImageProperties((prevState) => ({
+      ...prevState,
+      originalImage: event.target.files[0],
+    }));
+    setFileName(event.target.files[0].name);
+  }, []);
+
+  const handleZoom = useCallback((event: React.ChangeEvent<any>) => {
+    const scale = +event.target.value;
+    setImageProperties((prevState) => ({ ...prevState, scale }));
+  }, []);
+
+  const handleRotate = useCallback((direction: "left" | "right") => {
+    setImageProperties((prevState) => ({
+      ...prevState,
+      rotate:
+        direction === "left"
+          ? (prevState.rotate - 90) % 360
+          : (prevState.rotate + 90) % 360,
+    }));
+  }, []);
+
+  const handlePositionChange = useCallback(
+    (position: ImageProperties["position"]) => {
+      setImageProperties((prevState) => ({ ...prevState, position }));
+    },
+    []
+  );
+
+  const handleSubmit = useCallback(
+    (event: React.FormEvent) => {
+      event.preventDefault();
+      const canvasScaled: HTMLCanvasElement =
+        editorRef.current!.getImageScaledToCanvas();
+      canvasScaled.toBlob((blob: any) => {
+        setImageProperties((prevState) => ({
+          ...prevState,
+          croppedImage: window.URL.createObjectURL(blob),
+        }));
+        setImgToBeSubmitted(
+          new File([blob], fileName, {
+            type: blob.type,
+          })
+        );
+      });
+    },
+    [editorRef, fileName]
+  );
+
+  const handleConfirm = useCallback(async () => {
+    if (!isLoading) {
```

Comment:
Why `async` function when you do not have any `await` inside your function declaration?